ringvax/__init__.py

```diff
@@ -47,52 +47,45 @@ def run(self) -> None:
         # queue is pairs (t_exposed, infector)
         # start with the index infection
         infection_queue: List[tuple[float, Optional[str]]] = [(0.0, None)]
-        termination = {"criterion": "", "last_complete_generation": -1}
-
-        while (
-            len(infection_queue) > 0
-            and len(self.query_people()) < self.params["max_infections"]
-        ):
-            # find the person who is infected next. this ensures that, if we stop the
-            # simulation after some number of infections, the first infections were done
-            # first
+
+        passed_max_generations = False
+
+        while True:
+            # Check if we need to stop the loop
+            n_infections = len(self.query_people())
+
+            if len(infection_queue) == 0 and not passed_max_generations:
+                self.termination = "extinct"
+                break
+            elif len(infection_queue) == 0 and passed_max_generations:
+                self.termination = "max_generations"
+                break
```

Comment:
```suggestion
            if len(infection_queue) == 0:
                if not passed_max_generations:
                    self.termination = "extinct"
                else:
                    self.termination = "max_generations"
                break
```

A proposal that I think may avoid future Andy misreading this the same way past Andy misread it. "Stop if the queue is empty, and then check why it's empty"

Reply:
TIL you can do multiline suggestions

Reply:
I adopted this suggestion, and put some more color on this section, adding some comments, and using an `else` rather than the `continue`

let me know if this is clearer